patina_dxe_core/src/hw_interrupt_protocol.rs

```diff
@@ -451,20 +451,23 @@ impl HwInterruptProtocolHandler {
     }
 }
 
-#[derive(IntoComponent, Default)]
+#[derive(IntoComponent)]
 /// A component to install the two hardware interrupt protocols.
-pub(crate) struct HwInterruptProtocolInstaller;
+pub(crate) struct HwInterruptProtocolInstaller(GicBases);
 
 impl HwInterruptProtocolInstaller {
+    pub fn new(gic_bases: GicBases) -> Self {
+        Self(gic_bases)
+    }
+
     fn entry_point(
         self,
         interrupt_manager: Service<dyn InterruptManager>,
-        gic_bases: Config<GicBases>,
         boot_services: StandardBootServices,
     ) -> patina::error::Result<()> {
-        log::info!("GICv3 initializing {:x?}", (gic_bases.0, gic_bases.1));
+        log::info!("GICv3 initializing {:x?}", (self.0.0, self.0.1));
```

Comment:
Is this comment meant to have the GIC version hardcoded in it?

```suggestion
        log::info!("GIC initializing {:x?}", (self.0.0, self.0.1));
```

Applies to other references in the function.

Reply:
Updated generically to GIC